### PR TITLE
fix: switch to pushpad for notifications

### PR DESCRIPTION
**Pull request by schalkneethling**
This switches us to using the PushPad notification widget and removes the homegrown solution.

> NOTE: The correct project id needs to be added to src/index.html

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -25,6 +25,16 @@
   <link href="images/favicon.png" rel="shortcut icon" type="image/x-icon">
   <link href="images/webclip.png" rel="apple-touch-icon">
   <script src="https://kit.fontawesome.com/95e41092af.js" crossorigin="anonymous"></script>
+  <script>
+    (function (p, u, s, h, x) { p.pushpad = p.pushpad || function () { (p.pushpad.q = p.pushpad.q || []).push(arguments) }; h = u.getElementsByTagName('head')[0]; x = u.createElement('script'); x.async = 1; x.src = s; h.appendChild(x); })(window, document, 'https://pushpad.xyz/pushpad.js');
+
+    pushpad('init', 'ProjectID');
```
Comment (jbothma) [inline src/index.html:31]:
```suggestion
    pushpad('init', '7550');
```

File: src/index.html [hunk 2]
```diff
@@ -25,6 +25,16 @@
...
+  <script>
+    (function (p, u, s, h, x) { p.pushpad = p.pushpad || function () { (p.pushpad.q = p.pushpad.q || []).push(arguments) }; h = u.getElementsByTagName('head')[0]; x = u.createElement('script'); x.async = 1; x.src = s; h.appendChild(x); })(window, document, 'https://pushpad.xyz/pushpad.js');
+
+    pushpad('init', '7550');
+    pushpad('widget', {
+      promptTitle: 'Subscribe to MyMuni Notifications',
+      promptMessage: 'Subscribe to receive instant notifications of service disruptions and other important information.',
+      promptButtonColor: "#0094ff",
+    });
+  </script>
```
Comment (jbothma) [inline src/index.html:37]: I think we need to inject this using src/js/webflow/import.js so that this doesn't get overwritten in the next webflow export

Comment (schalkneethling) [reply src/index.html:37]: Indeed, completely forgot about that.

**Comment by jbothma** [comment]
@schalkneethling please check that you're happy with my additions due to this note:

Screenshot_2021-01-29_10-23-42

**Comment by schalkneethling** [comment]
> @schalkneethling please check that you're happy with my additions due to this note:
> 
> Screenshot_2021-01-29_10-23-42

So, I had that, but it sounded like if you have an existing service worker you do not need to inject that one, but only need to add the `updateViaCache` entry. I had a read [over this issue](https://github.com/GoogleChrome/workbox/issues/1519) on Workbox and it kinda sounded like they take care of it internally.

With that, I removed the `importScripts` and, testing it, it still seemed to work just fine(perhaps it was a fluke).
The `serviceWorkerPath` path docs is a little odd though:

> String serviceWorkerPath: the default location is '/service-worker.js'. If your service worker really needs to be in a different location you can set its path here, however remember that the location of the service worker can affect its behavior and you usually want to put it at the root of your website: don't put the service worker outside the root path unless you know exactly how service workers work. If your web app already registers the service worker, you should set this option to null, however in this case make sure that your application calls navigator.serviceWorker.register with the option updateViaCache: 'none', otherwise your notifications may stop working when we release new updates.

Specifically this piece:

> If your web app already registers the service worker, you should set this option to null, however in this case make sure that your application calls navigator.serviceWorker.register with the option updateViaCache: 'none', otherwise your notifications may stop working when we release new updates.

Your updates do make complete sense though, and I assume you do need to also inject their service worker. I thought so as well, but when I removed it and everything still worked, I thought the docs were saying

1. Inject this but,
2. if you have your own existing service worker, do the things as indicated at https://pushpad.xyz/docs/javascript_sdk_reference#pushpad_init_docs

**Comment by jbothma** [comment]
yeah - I suspect it might have worked accidentally, perhaps our push even handler was coincidentally compatible with theirs?

**Comment by schalkneethling** [comment]
> yeah - I suspect it might have worked accidentally, perhaps our push even handler was coincidentally compatible with theirs?

Perhaps 🤷‍♂️ 🙃

**Pull request merged**